analysis/filter/MergeFilter/src/main/kotlin/de/maibornwolff/codecharta/filter/mergefilter/LargeMerge.kt

```diff
@@ -0,0 +1,54 @@
+package de.maibornwolff.codecharta.filter.mergefilter
+
+import de.maibornwolff.codecharta.model.BlacklistItem
+import de.maibornwolff.codecharta.model.Edge
+import de.maibornwolff.codecharta.model.Node
+import de.maibornwolff.codecharta.model.NodeType
+import de.maibornwolff.codecharta.model.Project
+
+class LargeMerge {
+    companion object {
+        fun packageProjectInto(project: Project, prefix: String): Project {
+            val modifiedProject = Project(
+                project.projectName,
+                moveNodesIntoFolder(project.rootNode, prefix),
+                project.apiVersion,
+                addFolderToEdgePaths(project.edges, prefix),
+                project.attributeTypes,
+                project.attributeDescriptors,
+                addFolderToBlackListPaths(project.blacklist, prefix)
+            )
+            return modifiedProject
+        }
+
+        private fun moveNodesIntoFolder(root: Node, folderName: String): List<Node> {
+            val newRoot = Node(
+                name = folderName,
+                type = NodeType.Folder,
+                children = root.children
+            )
+            val mutableRoot = root.toMutableNode()
+            mutableRoot.children = mutableSetOf(newRoot.toMutableNode())
+            return listOf(mutableRoot.toNode())
```

Comment:
Isn't this equivalent to 
```
    val newRoot = Node(
        name = folderName,
        type = NodeType.Folder,
        children = listOf(root)
    )
    return listOf(newRoot)
    ```

Reply:
I don't think so, my approach: `root > folderName > root's children`
Your's: `folderName > root > root's children`

But the name `newRoot` is definitely not good. It is not a new root, it's just a folder inserted. At first, I used your approach, but that is not what the issue is about ^^

Reply:
Oh I see. Yeah the name confused me a bit. But that works then 👍